app/src/main/kotlin/com/brackeys/ui/feature/base/providers/ContextProvider.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2020 Brackeys IDE contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.brackeys.ui.feature.base.providers
+
+import android.content.ContentProvider
+import android.content.ContentValues
+import android.content.Context
+import android.database.Cursor
+import android.net.Uri
+
+/**
+ * The [Context] object is obtained without invasion, which reduces code coupling.
+ *
+ * @author gzu-liyujiang (1032694760@qq.com)
+ * @since 2020/12/10 10:06
+ */
+internal class ContextProvider : ContentProvider() {
+
+    override fun onCreate(): Boolean {
+        initApplicationContext(context!!)
```

Comment:
Can be replaced with `context?.let(::initApplicationContext)`

Reply:
Thanks, my kotlin is bad! :joy: